src/pages/admin/event/index.tsx

```diff
@@ -5,21 +5,45 @@ import { EventAPI } from '@/lib/api';
 import withAccessType from '@/lib/hoc/withAccessType';
 import { PermissionService } from '@/lib/services';
 import { PublicEvent } from '@/lib/types/apiResponses';
+import { Community } from '@/lib/types/enums';
 import style from '@/styles/pages/manage-events.module.scss';
+import { IconButton } from '@mui/material';
 import { GetServerSideProps, NextPage } from 'next';
-import Link from 'next/link';
-import { useMemo, useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { ChangeEvent, useEffect, useMemo, useState } from 'react';
+import { AiOutlineSearch } from 'react-icons/ai';
+import { BsFillGrid3X3GapFill } from 'react-icons/bs';
+import { FaChevronLeft, FaChevronRight, FaThList } from 'react-icons/fa';
 
 interface IProps {
   allEvents: PublicEvent[];
   pastEvents: PublicEvent[];
   futureEvents: PublicEvent[];
 }
 
+const NO_FILTER_OPTION = 'All Communities';
+const communityFilterOptions = [
+  NO_FILTER_OPTION,
+  ...(Object.keys(Community) as Array<keyof typeof Community>).map(key => Community[key]),
+];
+
+const communityMatch = (event: PublicEvent, filter: string): boolean => {
+  if (filter === NO_FILTER_OPTION) {
+    return true;
+  }
+  return event.committee === filter;
+};
+
 const ManageAllEventsPage: NextPage<IProps> = (props: IProps) => {
+  const router = useRouter();
   const { allEvents, pastEvents, futureEvents } = props;
 
   const [mode, setMode] = useState<'all' | 'past' | 'future'>('future');
+  const [query, setQuery] = useState<string>('');
+  const [viewMode, setViewMode] = useState<'card' | 'row'>('row');
+  const [communityFilter, setCommunityFilter] = useState<string>(NO_FILTER_OPTION);
+  const [currentPage, setCurrentPage] = useState<number>(0);
+  const [paginationSize, setPaginationSize] = useState<'All' | number>('All');
```

Comment:
Should it default to a smaller number than just all? If there's a lot of events, it probably wouldn't be great if the user had to wait for all the events to render before being able to select a smaller page size.

Reply:
The leaderboard PR #63 also adds pagination, searching, and filters. Perhaps some day we might want to merge them into a common component

Reply:
That makes sense, if you want you could refactor #63 to make some of the stuff common components and I can refactor this branch to use that!